### DISCUSSION TITLE
User Interface

### DISCUSSION THREAD

**Discussion by hay-kot** [mealie-recipes/mealie#229] [Ideas]
Share your ideas or comments about the user interface

**Comment by carlyman** [comment]
I really love how clean the interface is, even little things like clicking an ingredient or recipe step to mark them as done.  I like how you've focused on keeping things simple and clean!

Two interface updates I'd like:
1. In recipe view, be able to categorize/group steps and/or ingredients.  E.g., "Pie Crust", "Pie Filling".  Some recipes, especially when an ingredient is duplicated between them, makes it easier to follow when grouped. 
2. Print view.  While I think 90% of the time I'll be using my mobile to use the recipe, being able to print cleanly will come in handy.

**Comment by ltctceplrm** [comment]
I like the idea of pages but I don't see a way to sort the recipes on the page by recent/older or alphabetically, the old categories option in the sidebar did have that option and it was very handy.

The next part isn't really related to user interface but still about the pages, I've got an internet recipe category and a homemade recipe category, it'd be cool if you could have a page that has combination of categories basically a "Homemade" page that has homemade+dessert and homemade+meals and homemade+soups or something like that as sub divisions but they're just called Dessert, Meals and Soups.

Other than that I love the UI, it is very clean and easy to use.

Edit: The new UI on smartphones does seem to load much slower than it used to, scrolling through recipes is very stuttery now.

**Comment by describe19** [comment]
Sorry if this is more asking for help than feedback but...I've created a page but I don't see how to add any recipes to it. Is this what tags are for? A recipe tagged with the same tag as the page will show up on the page? If this is the case then for some reason I cannot add tags to pages. They won't save - as soon as I click the custom pages 'save' button in the bottom right, the coloured tag disappears from the page.

Here is a video of me trying (and failing) to set up a page: https://i.imgur.com/Cg6XUeA.mp4

**Reply by ltctceplrm** [reply]
This would probably be better if you opened an issue for it https://github.com/hay-kot/mealie/issues
On my install (docker) the custom pages are only for categories and not tags, I see that you only have Chicken as a category, so I'd try to add that one instead and see if that works (or just add a vegetarian category and try with that). Maybe do ctrl+F5 too after saving it to see if it shows up.

**Reply by ltctceplrm** [reply]
Ah I see that 0.4.1a changed the text for Categories to Tags in the custom page screen. But it seems like it still only works with categories and not tags, except that you can't add categories anymore, there's definitely a bug/oversight there. I've created an issue for it [here](https://github.com/hay-kot/mealie/issues/290)

**Comment by dshokouhi** [comment]
I have been working on adding a bunch of recipes and went back to make edits like adding categories and tags.  I keep forgetting to save the page when I navigate away. An auto save feature or "you have unsaved changes" prompt may help out. Not sure if its just me there.  Also would be nice to add the save button to the floating button so its more visible/apparent the user still needs to save.

**Reply by ikornaselur** [reply]
Just had this happen.. multiple times! Came here to see if anyone had mentioned it before.
In my opinion, the save button needs to be much more prevalent, and ideally, the unsaved data is stored at least locally so that if you navigate away and realise, opening edit should bring up the unsaved state. This is how it seems to work on GitHub, where if you are editing a PR for example and navigate away, when you come back to edit the description it will remember your edits.
